packages/astro/src/integration/index.ts

```diff
@@ -35,19 +35,30 @@ export const sentryAstro = (options: SentryOptions = {}): AstroIntegration => {
 
         // We don't need to check for AUTH_TOKEN here, because the plugin will pick it up from the env
         if (shouldUploadSourcemaps && command !== 'dev') {
-          // TODO(v9): Remove this warning
-          if (config?.vite?.build?.sourcemap === false) {
-            logger.warn(
-              "You disabled sourcemaps with the `vite.build.sourcemap` option. Currently, the Sentry SDK will override this option to generate sourcemaps. In future versions, the Sentry SDK will not override the `vite.build.sourcemap` option if you explicitly disable it. If you want to generate and upload sourcemaps please set the `vite.build.sourcemap` option to 'hidden' or undefined.",
-            );
+          const computedSourceMapSettings = getUpdatedSourceMapSettings(config, options);
+
+          let updatedFilesToDeleteAfterUpload: string[] | undefined = undefined;
+
+          if (
+            typeof uploadOptions?.filesToDeleteAfterUpload === 'undefined' &&
+            computedSourceMapSettings.previousUserSourceMapSetting === 'unset'
+          ) {
+            updatedFilesToDeleteAfterUpload = ['./dist/**/client/**/*.map', './dist/**/server/**/*.map'];
```

Comment:
m: In light of adapters possibly copying files outside of the `dist` directory, should we generalize the glob pattern? 

```suggestion
            updatedFilesToDeleteAfterUpload = ['*.map'];
```

Reply:
I was testing it with the Vercel adapter, and the source maps were deleted in the `.vercel` folder as well. I also checked with just deleting the files in the client and this also reflected to the `.vercel` folder.

Reply:
I'll write a comment about this to clarify it.